src/test/java/com/testapp/subject/controller/SubjectControllerIntegrationTest.java

```diff
@@ -19,14 +24,16 @@
 
 @RunWith(SpringRunner.class)
 @SpringBootTest(classes = Application.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+//@WebMvcTest(value = Store.class)
+//@ImportAutoConfiguration
 @AutoConfigureMockMvc
-class SubjectControllerIntegrationTest {
+public class SubjectControllerIntegrationTest {
 
     @Autowired
-    MockMvc mockMvc;
+    private MockMvc mockMvc;
 
     @Autowired
-    SubjectRepository subjectRepository;
+    private SubjectRepository subjectRepository;
 
     @Test
     public void when_user_is_not_authenticated_it_should_redirect_to_login() throws Exception {
```

Comment:
let's rename it to: `when_user_is_not_authenticated_the_list_page_should_be_redirected_to_login()`